infrastructure/deploy.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash -e
+
+# This script must be run from /infrastructure!
+
+# Make our IP address known to terraform.
+source ../common.sh
+export TF_VAR_host_ip=`wget -qO- http://ipecho.net/plain ; echo`
+
+# Copy ingress config to top level so it can be applied.
+cp deploy/ci_ingress.tf .
+
+# Output the plan for debugging deployments later.
+terraform plan
+
+# Open up ingress to AWS for Circle, stop jobs, migrate DB.
+echo "Applying ingress.."
+terraform apply -auto-approve
+
+# Find address of Nomad server.
+export NOMAD_LEAD_SERVER_IP=`terraform output nomad_server_1_ip`
+export NOMAD_ADDR=http://$NOMAD_LEAD_SERVER_IP:4646
+
+# This function checks what the status of the Nomad agent is.
+# Returns an HTTP response code. i.e. 200 means everything is ok.
+check_nomad_status () {
+    echo $(curl --write-out %{http_code} \
+                  --silent \
+                  --output /dev/null \
+                  $NOMAD_ADDR/v1/status/leader)
+}
+
+
+# Wait for Nomad to get started in case the server just went up for
+# the first time.
+echo "Confirming Nomad cluster.."
+start_time=$(date +%s)
+diff=0
```

Comment:
We're checking for the existence of the cluster, but we should be checking for the health of the _client_ or there could be a race condition on first deploy.